lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -157,7 +160,30 @@ export class BrowserCacheManager extends CacheManager {
      * @param value
      */
     setItem(key: string, value: string): void {
-        this.browserStorage.setItem(key, value);
+        try{
+            this.browserStorage.setItem(key, value);
+        }
+        catch(e){
+            // check for quota exceeding error 
+            if(e.code === BrowserAuthErrorMessage.chromiumStorageException.code || e.code === BrowserAuthErrorMessage.firefoxStorageException.code || e.name === "QuotaExceededError" || e.name === "NS_ERROR_DOM_QUOTA_REACHED")
+            {
+                this.logger.error("Could not access browser storage. This may be caused by exceeding the quota.");
```

Comment:
Optional. Since this message suggests that an error "may be caused by exceeding the quota" it could be easier just to check if an error message contains "quota" to avoid referencing to 3rd party error codes that can change in the future (`22` for chrome and `1014` for FF).